metricflow/test/model/modify/modify_manifest.py

```diff
@@ -0,0 +1,16 @@
+from __future__ import annotations
+
+from dbt_semantic_interfaces.implementations.semantic_manifest import PydanticSemanticManifest
+from dbt_semantic_interfaces.transformations.semantic_manifest_transformer import PydanticSemanticManifestTransformer
+from dbt_semantic_interfaces.transformations.transform_rule import SemanticManifestTransformRule
+
+
+def modify_manifest(
```

Comment:
We should be a bit careful with these. In this case it's responsible, so I think that's fine, but it's very easy to make tests that have so much dynamic configuration they become hard to reason about. The advantage of written-out manifest inputs is they're easier to read and more aligned with the product behavior itself - that's why we got rid of the tests where we were updating config objects in memory in the first place.

We may want to add some more documentation to this helper to indicate when it's generally acceptable to use it. I think any case where we need to update a highly specific element - like the where filter - in a way that is widespread and mechanical so we can see the differences in behavior is fine.

Reply:
I agree about limiting the use cases. Updated the comment.